library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplayConfiguration.kt

```diff
@@ -53,10 +58,17 @@ internal constructor(
             return this
         }
 
+        protected fun setViewMapper(
+                type:Class<*>,
+                mapper: WireframeMapper<View,*>) {
+                this.customViewMappers[type] = mapper
+        }
```

Comment:
yes I went this way first part even further `fun <T:View, S:Wireframe> setViewMapper(type:Class<T>, mapper:WireframeMapper<T,S>)`
but problem is that I have the map declared as:
` internal val customViewMappers: Map<Class<*>, WireframeMapper<View, *>> = emptyMap()`
and Kotlin as Java is stupid and doesn't know how to map `T` to `View` and `S` to `*` which in Kotlin means `? extends Wireframe`. So I will end up with force casting it but I guess if we go with the public api method I will better go this way. It will force the user to provide the right mapper for the right type at least.